.github/workflows/cmake.yml

```diff
@@ -150,7 +150,7 @@ jobs:
     - name: Print target logs
       if: ${{ !cancelled() && matrix.debug-macros == 'NORMAL_DEBUG'}}
       working-directory: ${{ github.workspace }}/Build
-      run: cat ./debug.log
+      run: '[ ! -e debug.log ] || cat debug.log'
```

Comment:
Hmm the point of this step is to make sure that we can see the boot dbg log even if the previous step which runs on target tests fails. If we don't generate the file at all because of "reasons" (maybe someone changes the run.sh step but forgot to update CI?) then this change would hide that, right? Maybe we could ferry some input/output from the `run tests` step? Not sure that's very elegant...

Reply:
You're right that this might "hide" an error if it fails at just the right moment. However, `run.sh` calls qemu with `-debugcon file:debug.log`, so this step has to fail extremely early for the file to be missing; in which case the emulation step will have generated interesting output and resulted in nonzero exit-code.

I'm not aware of a much better solution:
- If we fail here, then it "hides" the problem by being duplicious.
- If we add `touch debug.log` before the emulation, then this step will also "fail silently".

What do you suggest?

Reply:
Hmm. Yeah we'd have to be calling qemu incorrectly in order for it to hide the failure. Could make it not a separate step but just `ninja run || cat ./debug.log` with some `echo startgroup` magic in the middle? It's probably fine as is, that kind of error should be caught pretty quickly by a reviewer or by someone checking the logs.

Reply:
I like that idea!